<a href="https://colab.research.google.com/github/valeriaramirezjimenez10-crypto/Analitica-de-negocios/blob/main/Reto_Naive_Bayes.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**RETO 1: LAVADO DE ACTIVOS**

En este documento se presenta el desarrollo y análisis de un modelo de Naive Bayes, cuyo objetivo es fortalecer el sistema analítico de una entidad financiera para la detección temprana de clientes con alto riesgo, específicamente en actividades relacionadas con el lavado de activos.

Para ello, se consideran las siguientes variables:

Lavado de Activos: Indica si el cliente ha sido previamente identificado en procesos relacionados con este delito.

Crecimiento patrimonial (%): Permite evaluar variaciones significativas o inusuales en el patrimonio de los clientes.

Monto transado: Refleja el volumen de operaciones financieras y resulta clave para comprender el perfil transaccional del cliente.

Ingresos y gastos: Estas variables facilitan la identificación de posibles inconsistencias entre entradas y salidas de dinero, además de contribuir a la caracterización del comportamiento financiero del cliente.

0. Carga de librerias

In [ ]:
import numpy as np #Numerica por excelencia
import pandas as pd #Comunicación con los archivos de excel
#Librerias especificas: Ej. Naive Bayes
from sklearn.naive_bayes import GaussianNB
from sklearn.metrics import confusion_matrix

1. Cargar la base de datos

In [ ]:
nxl="/content/2. LavadoActivos.xlsx"
XDB=pd.read_excel(nxl,sheet_name=0)
Xd=XDB[['ingresos_usd','gastos_usd','monto_transado_tarjetas_usd','porcentaje_crecimiento_patrimonio']]
Xd.head()
yd=XDB['lavado_activos']
yd.head()

,lavado_activos
0,1
1,1
2,0
3,0
4,0


2. Implementación del Modelo Naive Bayes

In [ ]:
mnb=GaussianNB()
mnb.fit(Xd,yd)
u=mnb.theta_
sigma=mnb.var_;sigma=np.sqrt(sigma)
print("'ingresos_usd','gastos_usd','monto_transado_tarjetas_usd','porcentaje_crecimiento_patrimonio")
print(u)
print("Las desviaciones son")
print(sigma)

'ingresos_usd','gastos_usd','monto_transado_tarjetas_usd','porcentaje_crecimiento_patrimonio
[[6.69073255e+03 3.60174970e+03 3.09046404e+03 4.46871435e+01]
 [7.02623043e+04 3.87072844e+04 3.38356197e+04 4.48208630e+01]]
Las desviaciones son
[[6.20696435e+03 3.21901567e+03 2.45685863e+03 1.41731672e+01]
 [1.13412265e+05 6.59025571e+04 6.29913911e+04 1.54786058e+01]]


**Análisis de resultados:**
A partir de los resultados obtenidos, se evidencia una diferencia significativa en los comportamientos financieros de los clientes previamente clasificados como de alto riesgo en lavado de activos, tanto en sus ingresos como en sus gastos. Estas variaciones permiten poner en duda el origen de montos tan elevados.
Adicionalmente, el volumen de transacciones realizadas con tarjeta también es mayor en los clientes con perfil previamente clasificado como alto riesgo, lo cual resulta coherente con los hallazgos anteriores.

Por otra parte, las desviaciones observadas en la categoría de alto riesgo superan a las del grupo de bajo riesgo, lo que indica que los clientes preclasificados como riesgosos presentan una mayor variabilidad en su comportamiento financiero, siendo menos estables, homogéneos y predecibles en comparación con aquellos de bajo riesgo.

3. Se procede con la evaluacion del modelo

In [ ]:
ydp=mnb.predict(Xd)
cm=confusion_matrix(yd,ydp)
print(cm)
VN=cm[0,0];FP=cm[0,1];FN=cm[1,0]; VP=cm[1,1];TDatos=len(XDB)
#1:Exactitud: Funcionamiento general del modelo
EX=(VP+VN)/TDatos
print("Exactitud: ",EX)
#2: Tasa de error: Porcentaje de fallo del sistema
TEr=(FP+FN)/TDatos
print("Tasa de error: ",TEr)
#3:Sensibilidad: Como se comporto con respecto a los positivos solamente de la base de datos
Se=VP/(VP+FN)
print("Sensibilidad: ",Se)
#4:Especificidad
Es=VN/(VN+FP)
print("Especificidad: ",Es)
#5. Precisión: Version de como funciona frente a los positivos solamente
Pr=VP/(VP+FP)
print("Precision: ",Pr)
#6: Predicción Negativa: Funcionamiento del modelo pronosticando creditos Pre Negados
Prn=VN/(VN+FN)
print("Predicción Negativa: ",Prn)

[[2179   44]
 [ 142  785]]
Exactitud:  0.940952380952381
Tasa de error:  0.05904761904761905
Sensibilidad:  0.8468176914778857
Especificidad:  0.9802069275753487
Precision:  0.9469240048250904
Predicción Negativa:  0.9388194743644981


**Analisis de resultados**
De acuerdo con los resultados, podemos observar de manera general que el modelo alcanzó con **exatitud** por encima de 94% en promedio, lo que indica en terminos generales el buen comportamiento del modelo frente a la clasificación de los clientes de alto riesgo y los sin riesgos, solo con una **tasa de error** del 5%. Se destaca **la sensibilidad** con un 84.6% lo que indica el buen funcionamiento de detección del modelo frente a identificar correctamente los clientes de alto riesgo de lavado de activos que si son de alto riesgo.
**La especificidad** tenemos un 98% de que tan estricto es el modelo, de cuales clientes dejo pasar que en relidad no presentaban ningun riesgo.
**Precision**: Con un 94,6% podemos ver el buen comportamieto del modelo ya que es altamente precioso en cuales clientes marca como riesgos que realmente si son riegosos.
**Prediccion negativa**: nos confirma que es un modelo confiable con un 93%.